doc/src/manual/style-guide.md

```diff
@@ -164,6 +164,62 @@ uses (e.g. `a[i]::Int`) than to try to pack many alternatives into one type.
 If a function name requires multiple words, consider whether it might represent more than one
 concept and might be better split into pieces.
 
+## Write functions that use similar argument precedence as Julia's `base/`
+
+As a general rule, the Base library uses the following order of arguments to functions,
+as applicable:
+
+1. **Function argument**.
+   This permits the use of `do` blocks for passing multiline anonymous functions.
+
+2. **I/O stream**.
+   Specifying the `IO` object first permits passing the function to functions such as
+   [`sprint`](@ref), e.g. `sprint(show, x)`.
+
+3. **Input being mutated**.
+   In [`fill!(x, v)`](@ref fill!), `x` is the object being mutated, and it appears before
+   the value to be inserted into `x`.
+
+4. **Type**.
+   Passing a type typically means that the output will have the given type.
+   In [`read(io, String)`](@ref read), the `IO` argument appears before the type.
+
+5. **Input not being mutated**.
+   In `fill!(x, v)`, `v` is *not* being mutated and it comes after `x`.
+
+6. **Key**.
```

Comment:
example?

Reply:
What the heck?? I could have _sworn_ I had an example here. I even remember what I had written (assuming I didn't imagine it).

Reply:
I have it locally so I guess I just forgot to push it.